File: SupermarketInventory/java/SupermarketInventory.java
public class SupermarketInventory {

    public static int getBestDayForProduct(int[][] productDetails, int productNumber) {

        int highestUnit = productDetails[productNumber][0];
        int highestUnitIndex = 0;

        for (int index = 0; index < productDetails[productNumber].length; index++) {

            int unit = productDetails[productNumber][index];

            if (unit > highestUnit) {

                highestUnit = unit;

                highestUnitIndex = index;
            }
        }

        return highestUnitIndex + 1;
    }

    public static int[] getBestSellingProduct(int[][] productDetails) {

        int[] bestSellingProduct = new int[2];

        int highestTotalUnit = Integer.MIN_VALUE;

        int highestUnitIndex = 0;

        for (int index = 0; index < productDetails.length; index++) {

            int totalUnit = 0;

            for (int innerIndex = 0; innerIndex < productDetails[0].length; innerIndex++) {

                totalUnit += productDetails[index][innerIndex];
            }

            if (totalUnit > highestTotalUnit) {

                highestTotalUnit = totalUnit;

                highestUnitIndex = index;
            }
        }

        bestSellingProduct[0] = highestUnitIndex + 1;

        bestSellingProduct[1] = highestTotalUnit;

        return bestSellingProduct;
    }

    public static int[] getLowestSellingProduct(int[][] productDetails) {

        int[] lowestSellingProduct = new int[2];

        int lowestTotalUnit = Integer.MAX_VALUE;

        int lowestUnitIndex = 0;

        for (int index = 0; index < productDetails.length; index++) {

            int totalUnit = 0;

            for (int innerIndex = 0; innerIndex < productDetails[0].length; innerIndex++) {

                totalUnit += productDetails[index][innerIndex];
            }

            if (totalUnit < lowestTotalUnit) {

                lowestTotalUnit = totalUnit;

                lowestUnitIndex = index;
            }
        }

        lowestSellingProduct[0] = lowestUnitIndex + 1;

        lowestSellingProduct[1] = lowestTotalUnit;

        return lowestSellingProduct;
    }

    public static int getOverallUnitSold(int[][] productDetails) {

        int totalUnit = 0;

        for (int index = 0; index < productDetails.length; index++) {

            for (int unit : productDetails[index]) {

                totalUnit += unit;
            }

        }

        return totalUnit;
    }

    public static void displaySalesSummary(int[][] productDetails) {

        System.out.println("\n========== STORE SALES SUMMARY ==========\n");

        System.out.printf("%-12s", "PRODUCT");

        for (int index = 0; index < productDetails[0].length; index++) {

            System.out.printf("DAY%-5d", index + 1);

        }

        System.out.printf("%-8s%s%n", "TOTAL", "BEST DAY");

        System.out.println("-------------------------------------------------------");

        for (int productIndex = 0; productIndex < productDetails.length; productIndex++) {

            System.out.printf("Product %-4d", productIndex + 1);

            int totalUnit = 0;

            for (int daysIndex = 0; daysIndex < productDetails[0].length; daysIndex++) {

                int unit = productDetails[productIndex][daysIndex];

                totalUnit += unit;

                System.out.printf("%-8d", unit);

            }

            int productBestDay = getBestDayForProduct(productDetails, productIndex);

            System.out.printf("%-8dDay %d%n", totalUnit, productBestDay);

        }

        System.out.println("-------------------------------------------------------\n");

        int[] bestSellingProduct = getBestSellingProduct(productDetails);

        int[] lowestSellingProduct = getLowestSellingProduct(productDetails);

        int overallUnitSold = getOverallUnitSold(productDetails);

        System.out.printf("""
        Best-selling product: Product %d (%d units)
        Lowest-selling product: Product %d (%d units)
        Overall units sold: %d
        """,
        bestSellingProduct[0],
        bestSellingProduct[1],
        lowestSellingProduct[0],
        lowestSellingProduct[1],
        overallUnitSold
        );

    }
}

File: SupermarketInventory/python/supermarket_inventory.py

import sys

class SupermarketInventory:

    def __init__(self, product_details):

        self.product_details = product_details


    def get_best_day(self, product_number):

        highest_unit = 0

        highest_unit_index = 0

        index = 1

        for unit in self.product_details[product_number]:

            if unit > highest_unit:

                highest_unit = unit

                highest_unit_index = index

            index += 1

        return highest_unit_index


    def get_best_selling_product_details(self):

        highest_total_unit = -sys.maxsize - 1

        highest_total_unit_index = 0

        for index in range(len(self.product_details)):

            total_unit = 0

            for unit in self.product_details[index]:

                total_unit += unit

            if total_unit > highest_total_unit:

                highest_total_unit = total_unit

                highest_total_unit_index = index + 1

        return [highest_total_unit_index, highest_total_unit]


    def get_lowest_selling_product_details(self):

        lowest_total_unit = sys.maxsize

        lowest_total_unit_index = 0

        for index in range(len(self.product_details)):

            total_unit = 0

            for unit in self.product_details[index]:

                total_unit += unit

            if total_unit < lowest_total_unit:

                lowest_total_unit = total_unit

                lowest_total_unit_index = index + 1

        return [lowest_total_unit_index, lowest_total_unit]

    def get_overall_total_units(self):

        total_unit = 0

        for index in range(len(self.product_details)):

            for unit in self.product_details[index]:

                total_unit += unit

        return total_unit

# math.inf